xline/src/storage/lease_store/mod.rs

```diff
@@ -56,133 +47,14 @@ where
-            } else {
-                break;
-            }
-        }
-        expired_leases
-    }
-
-    /// Renew lease
-    fn renew(&mut self, lease_id: i64) -> Result<i64, ExecuteError> {
-        self.lease_map.get_mut(&lease_id).map_or_else(
-            || Err(ExecuteError::lease_not_found(lease_id)),
-            |lease| {
-                if lease.expired() {
-                    return Err(ExecuteError::lease_expired(lease_id));
-                }
-                let expiry = lease.refresh(Duration::default());
-                let _ignore = self.expired_queue.update(lease_id, expiry);
-                Ok(lease.ttl().as_secs().cast())
-            },
-        )
-    }
-
-    /// Attach key to lease
-    fn attach(&mut self, lease_id: i64, key: Vec<u8>) -> Result<(), ExecuteError> {
-        self.lease_map.get_mut(&lease_id).map_or_else(
-            || Err(ExecuteError::lease_not_found(lease_id)),
-            |lease| {
-                lease.insert_key(key.clone());
-                let _ignore = self.item_map.insert(key, lease_id);
-                Ok(())
-            },
-        )
-    }
-
-    /// Detach key from lease
-    fn detach(&mut self, lease_id: i64, key: &[u8]) -> Result<(), ExecuteError> {
-        self.lease_map.get_mut(&lease_id).map_or_else(
-            || Err(ExecuteError::lease_not_found(lease_id)),
-            |lease| {
-                lease.remove_key(key);
-                let _ignore = self.item_map.remove(key);
-                Ok(())
-            },
-        )
-    }
-
-    /// Check if a lease exists
-    fn contains_lease(&self, lease_id: i64) -> bool {
-        self.lease_map.contains_key(&lease_id)
-    }
-
-    /// Grant a lease
-    fn grant(&mut self, lease_id: i64, ttl: i64, is_leader: bool) -> PbLease {
-        let mut lease = Lease::new(lease_id, ttl.max(MIN_LEASE_TTL).cast());
-        if is_leader {
-            let expiry = lease.refresh(Duration::ZERO);
-            let _ignore = self.expired_queue.insert(lease_id, expiry);
-        } else {
-            lease.forever();
-        }
-        let _ignore = self.lease_map.insert(lease_id, lease.clone());
-        PbLease {
-            id: lease.id(),
-            ttl: lease.ttl().as_secs().cast(),
-            remaining_ttl: lease.remaining_ttl().as_secs().cast(),
-        }
-    }
-
-    /// Revokes a lease
-    fn revoke(&mut self, lease_id: i64) -> Option<Lease> {
-        self.lease_map.remove(&lease_id)
-    }
-
-    /// Demote current node
-    fn demote(&mut self) {
-        self.lease_map.values_mut().for_each(Lease::forever);
-        self.expired_queue.clear();
-    }
-
-    /// Promote current node
-    fn promote(&mut self, extend: Duration) {
-        for lease in self.lease_map.values_mut() {
-            let expiry = lease.refresh(extend);
-            let _ignore = self.expired_queue.insert(lease.id(), expiry);
-        }
-    }
-}
-
 /// Lease store inner
 #[derive(Debug)]
 pub(crate) struct LeaseStoreBackend<DB>
```

Comment:
`LeaseStoreBackend` is only used in this file, please remove `pub(crate)`

Reply:
Removed